pkg/steps/index_generator.go

```diff
@@ -63,6 +62,7 @@ func (s *indexGeneratorStep) Run(ctx context.Context) error {
 }
 
 func (s *indexGeneratorStep) run(ctx context.Context) error {
+	logrus.Warn("Building index images is deprecated and will be removed from ci-operator soon")
```

Comment:
Maybe this needs to be a little more noticeable:
```suggestion
	logrus.Warn("DEPRECATION WARNING: Building index images is deprecated and will be removed from ci-operator soon")
```